app/controllers/v0/api_controller.rb

```diff
@@ -2,13 +2,26 @@
 module V0
   class ApiController < ApplicationController
     skip_before_action :authenticate_user!
+    before_action :resolve_version
 
     def cors_preflight
       head(:ok)
     end
 
     private
 
+    # Newest production data version assumed when version param is undefined
+    def resolve_version
+      v = params[:version]
+      version = v.present? ? Version.find_by(number: v) : Version.default_version
+      raise ActiveRecord::RecordNotFound, "Version #{v} not found" unless version.try(:number)
```

Comment:
This might be better as a Common::Exceptions::RecordNotFound, or even better Common::Exceptions::InvalidFieldValue since it's a parameter that can't be found as opposed to a top-level resource.